lib/src/generator/generator_utils.dart

```diff
@@ -13,31 +13,35 @@ import 'package:dartdoc/src/model/model_element.dart';
 /// Convenience function to generate category JSON since different generators
 /// will likely want the same content for this.
 String generateCategoryJson(Iterable<Categorization> categories, bool pretty) {
-  var encoder = pretty ? JsonEncoder.withIndent(' ') : JsonEncoder();
   // ignore: omit_local_variable_types
-  final List<Map<String, Object>> indexItems =
-      categories.map((Categorization e) {
-    var data = <String, Object>{
-      'name': e.name,
-      'qualifiedName': e.fullyQualifiedName,
-      'href': e.href,
-      'type': e.kind,
-    };
+  final List<Map<String, Object>> indexItems = categories
+      .map((Categorization cat) {
+        final data = <String, Object>{
+          'name': cat.name,
+          'qualifiedName': cat.fullyQualifiedName,
+          'href': cat.href,
+          'type': cat.kind,
+        };
 
-    if (e.hasCategoryNames) data['categories'] = e.categoryNames;
-    if (e.hasSubCategoryNames) data['subcategories'] = e.subCategoryNames;
-    if (e.hasImage) data['image'] = e.image;
-    if (e.hasSamples) data['samples'] = e.samples;
-    return data;
-  }).toList();
+        if (cat.hasCategoryNames) {
+          data['categories'] = cat.categoryNames;
+        }
+        if (cat.hasSubCategoryNames) {
+          data['subcategories'] = cat.subCategoryNames;
+        }
+        if (cat.hasImage) {
+          data['image'] = cat.image;
+        }
+        if (cat.hasSamples) {
+          data['samples'] = cat.samples;
+        }
+        return data;
+      })
+      .sorted(_sortElements)
+      .toList(growable: false);
```

Comment:
Given that `.sorted` already gives us a `List`, the advantage to transforming a local variable to `growable: false` here seems small as the only thing we're doing with it is passing it to the encoder.